manifests/init.pp

```diff
@@ -5,27 +5,112 @@
+#   KSM_SLEEP_MSEC * 16 / Total GB of RAM The final sleep value will be
+#   written to /sys/kernel/mm/ksm/sleep_millisecs
+#
+# @param ksm_npages_boost
+#   Amount to increment the number of pages to scan. The number of pages to
+#   be scanned will be increased by KSM_NPAGES_BOOST when the amount of free
+#   ram < threshold (see KSM_THRES_* below)
+#
+# @param ksm_npages_decay
+#   Amount to decrease the number of pages to scan The number of pages to be
+#   scanned will be decreased by KSM_NPAGES_DECAY when the amount of free ram
+#   >= threshold (see KSM_THRES_* below)
+#
+# @param ksm_npages_min
+#   Minimum number of pages to be scanned at all times
+#   If this variable is set to 'shmall', then half of the value in
+#   /proc/sys/kernel/shmall will be used.
+#
+# @param ksm_npages_max
+#   Maximum number of pages to be scanned at all times
+#   If this variable is set to 'shmall', then the value in
+#   /proc/sys/kernel/shmall will be used.
+#
+# @param ksm_thres_coef
+#   If free memory is less than this percentage KSM will be activated.
+#   NOTE: Only KSM_THRES_CONST or KSM_THRES_COEF is actually used.  Whichever
+#   results in a larger number wins.
+#
+# @param ksm_thres_const
+#   If free memory is less than this number KSM will be activated
+#   NOTE: Only KSM_THRES_CONST or KSM_THRES_COEF is actually used.  Whichever
+#   results in a larger number wins.
+#
+# @param kvm_package_list List of packages related to kvm to be managed
+# @param manage_sysctl Disable to avoid setting sysctl settings in this module.
+# @param package_ensure Ensure setting for all packages in this module
+#
+# @author https://github.com/simp/pupmod-simp-libvirt/graphs/contributors
+#
+class libvirt (
+  Array[String] $package_list,
+  String $service_ensure,
+  Boolean $ksm,
+  Boolean $kvm,
+  Boolean $load_kernel_modules,
+
+  # KSM
+  Optional[Integer]               $ksm_max_kernel_pages,
+  Integer                         $ksm_monitor_interval,
+  Integer                         $ksm_sleep_msec,
+  Integer                         $ksm_npages_boost,
+  Integer[default,0]              $ksm_npages_decay,
+  Variant[Enum['shmall'],Integer] $ksm_npages_min,
+  Variant[Enum['shmall'],Integer] $ksm_npages_max,
+  Integer                         $ksm_thres_coef,
+  Optional[Integer]               $ksm_thres_const,
 
-  if $facts['operatingsystem'] in ['RedHat', 'CentOS'] {
-    $package_list = $facts['operatingsystemmajrelease'] ? {
-      '7'     => ['libvirt', 'virt-viewer', 'virt-install'],
-      default => ['libvirt', 'virt-viewer', 'python-virtinst']
+  # KVM
+  Array[String] $kvm_package_list,
+  Boolean $manage_sysctl,
+
+  String $package_ensure = simplib::lookup('simp_options::package_ensure', { 'default_value' => 'installed' }),
+) {
+
+  if $kvm { include 'libvirt::kvm' }
+
+  # This is a workaround until we can afford a breaking change to disallow
+  # calling the ksm class by itself
+  if $ksm {
+    class { 'libvirt::ksm':
```

Comment:
This will break a direct `include` of the class which is allowed by the nature of `ksm`. Please remove.